addons/mail/models/res_partner.py

```diff
@@ -135,6 +135,13 @@ def _get_channels_as_member(self):
                 ('is_pinned', '=', True),
             ])),
         ])
+        # get the 'group chat' channels
+        channels |= self.env['mail.channel'].search([
+            ('channel_type', '=', 'group'),
+            ('channel_last_seen_partner_ids', 'in', self.env['mail.channel.partner'].sudo()._search([
+                ('partner_id', '=', self.id),
+            ])),
```

Comment:
Wouldn't `channel_last_seen_partner_ids.partner_id = self.id` work just as well and avoid the need for a sudo?

Reply:
Probably would be possible indeed, this was a copy paste of the above conditions where additional "where" parts were needed in the join, but as there is only the join with no extra conditions for this one it should be possible.